src/mds/CDir.cc

```diff
@@ -1850,6 +1850,10 @@ CDentry *CDir::_load_dentry(
       }
     }
 
+    // Always parse the CDentry's fscrypted alternate_name from the end.
+    if (!q.end())
+      decode(dn->alternate_name, q);
```

Comment:
this will prevent us from adding any field to InodeStoreBase. (see InodeStoreBase::decode_bare()). I suggest adding new type tags for dentry (current ones are 'I' and 'L') and encode primary/remote link with proper version handling

Reply:
Yeah, it makes sense, will fix it. Thanks.